src/NewPost.js

```diff
@@ -0,0 +1,100 @@
+import React, { Component } from "react";
+import { data } from "./data";
+
+export class NewPost extends Component {
+  state = {
+    title: "",
+    author: "",
+    createdOn: "",
+    body: "",
+    tags: ""
+  };
+
+  createDateOnSubmit = () => {
+    const date = new Date();
+    let year = date.getFullYear();
+    let month = date.getMonth();
+    let day = date.getDate();
+    const dateNow = `${year}-${month + 1}-${day}`;
+    return dateNow;
+  };
+
+  createUniqueIdOnSubmit = () => {
+    const id = data.length;
+    return id;
+  };
+
+  convertTagsToArray = tags => {
+    const splitOn = ",";
+    let resultArray = tags.split(splitOn);
+    resultArray = resultArray.map(function(str) {
+      return str.trim();
+    });
+
+    return resultArray;
+  };
+
+  handleChange = e => {
+    this.setState({
+      [e.target.name]: e.target.value
+    });
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+
+    const resultArray = this.convertTagsToArray(this.state.tags);
+    const blogPost = Object.assign({}, this.state);
+    blogPost.tags = resultArray;
+
+    const newId = this.createUniqueIdOnSubmit(this.state.id);
+    blogPost.id = newId;
+
+    const todaysDate = this.createDateOnSubmit(this.state.createdOn);
+    blogPost.createdOn = todaysDate;
+
+    data.push(blogPost);
```

Comment:
let's extract this to a `saveBlogPost` function that takes `blogPost` and handles the details of saving it. then we can replace the `push` to a database call later without changing anything here in the component